packages/flutter/lib/src/gestures/recognizer.dart

```diff
@@ -634,7 +636,7 @@ abstract class PrimaryPointerGestureRecognizer extends OneSequenceGestureRecogni
   /// Drifting past the allowed slop amount causes the gesture to be rejected.
   ///
   /// Can be null to indicate that the gesture can drift for any distance.
-  /// Defaults to 18 logical pixels.
+  /// Defaults to touch slop from [gestureSettings] (18 logical pixels).
```

Comment:
Is this referring to MediaQueryData.gestureSettings? Where is that being accessed here?

Reply:
Yes. Line 658 in this commit:

`double get _defaultTouchSlop => gestureSettings.touchSlop ?? kTouchSlop;`

(`gestureSettings` comes from base class)

Reply:
Where does it assign the inherited MediaQueryData.gestureSettings?

Reply:
The 2 getters immediately before that (`_preAcceptSlopTolerance` and `_postAcceptSlopTolerance`) default to it (by using `_defaultTouchSlop`) UNLESS other tolerances are specified (including null)

![image](https://github.com/user-attachments/assets/393747e2-3235-4ff0-9277-68815ad39642)

Reply:
Ah ok, I think I misunderstood the description above. Here, in the super class, and the implementing classes, gestureSettings is not actually set. It looks like only once up in GestureDetector is MediaQuery factored in.

Reply:
In this case. would update the docs to not say it comes from gesture settings, since they could be null. There is a default value here still based on kTouchSlop. :)

Reply:
Yes. So the intent of this PR is that instead of always using `_kTouchSlop`, `PrimaryPointerGestureRecognizer` will use these values for pre/postAcceptSlopTolerance (in this order of priority):

- Values passed into constructor parameters
- Value read from `MediaQuery.of(context).gestureSettings`
- `_kTouchSlop`

Reply:
I understand. But this PR does not do this.
`PrimaryPointerGestureRecognizer` does not at any point, read from `MediaQuery.of(context).gestureSettings`, and neither does its super class. How does `PrimaryPointerGestureRecognizer` access `MediaQuery.of(context).gestureSettings`?

Reply:
`GestureDetector` widget reads `MediaQuery.maybeGestureSettingsOf(context)` in its build method and sets it for each recognizer instance.

https://github.com/flutter/flutter/blob/554c814ada6419c4ece39a8de5ff38e16cda614b/packages/flutter/lib/src/widgets/gesture_detector.dart#L1051C34-L1051C49

My understanding is that for the most common usages, the classes I'm modifying in this PR already have access to `gestureSettings` (from their parent classes and `GestureDetector`). They're just not doing anything with those values internally and always using `kTouchSlop`.

Reply:
Thanks for explaining. I am referring to the description of this PR:

> Use MediaQuery.gestureSettings.touchSlop in PrimaryPointerGestureRecognizer.handleEvent() unless a non-default value (including null) is specified.

This is not true of this PR technically. This is the point I was trying to make. Unless the user explicitly sets it on an underlying subclass, it will not be used.

Reply:
Good point. Removed `MediaQuery` from the PR description. 👍